client/state/sites/reducer.js

```diff
@@ -269,6 +273,51 @@ export const items = withSchemaValidation( sitesSchema, ( state = null, action )
 	return state;
 } );
 
+/**
+ * Tracks all known site excerpt objects, indexed by site ID.
+ *
+ * @param  {object} state  Current state
+ * @param  {object} action Action payload
+ * @returns {object}        Updated state
+ */
+export const excerpts = withSchemaValidation( siteExcerptsSchema, ( state = null, action ) => {
+	if ( state === null && action.type !== SITE_EXCERPTS_RECEIVE ) {
+		return null;
+	}
+	switch ( action.type ) {
+		case SITE_EXCERPTS_RECEIVE: {
+			// Normalize incoming site(s) to array
+
+			const sites = action.site ? [ action.site ] : action.sites;
+
+			// SITE_EXCERPTS_RECEIVE occurs when we receive the entire set of user
+			// sites (replace existing state). Otherwise merge into state.
+
+			const initialNextState = SITE_EXCERPTS_RECEIVE === action.type ? {} : state;
+
+			return reduce(
+				sites,
+				( memo, site ) => {
+					// Bypass if site object hasn't changed
+					if ( isEqual( memo[ site.ID ], site ) ) {
+						return memo;
+					}
+
+					// Avoid mutating state
+					if ( memo === state ) {
+						memo = { ...state };
+					}
+
+					memo[ site.ID ] = site;
+					return memo;
+				},
+				initialNextState || {}
+			);
+		}
+	}
+	return state;
+} );
+
```

Comment:
Should we invalidate, and potentially refresh, `state.sites.excerpts` when `state.sites.items` is updated?